service/gov_pay.go

```diff
@@ -135,6 +145,86 @@ func (gp *GovPayService) GetGovPayPaymentDetails(paymentResource *models.Payment
 	return paymentDetails, Success, nil
 }
 
+// GetGovPayRefundSummary gets refund summary of a GovPay payment
+func (gp *GovPayService) GetGovPayRefundSummary(req *http.Request, id string) (*models.PaymentResourceRest, *models.RefundSummary, ResponseType, error) {
+	// Get PaymentSession for the GovPay call
+	paymentSession, response, err := gp.PaymentService.GetPaymentSession(req, id)
+	if err != nil {
+		err = fmt.Errorf("error getting payment resource: [%v]", err)
+		log.ErrorR(req, err)
+		return nil, nil, response, err
+	}
+
+	govPayResponse, err := callGovPay(gp, paymentSession)
+	if err != nil {
+		err = fmt.Errorf("error getting payment information from gov pay: [%v]", err)
+		log.ErrorR(req, err)
+
+		return nil, nil, Error, err
+	}
+
+	switch govPayResponse.RefundSummary.Status {
+	case RefundUnavailable:
+		err = errors.New("cannot refund the payment - check if the payment failed")
+		return nil, nil, InvalidData, err
+	case RefundFull:
+		err = errors.New("cannot refund the payment - the full amount has already been refunded")
+		return nil, nil, InvalidData, err
+	case RefundPending:
+		err = errors.New("cannot refund the payment - the user has not completed the payment")
+		return nil, nil, InvalidData, err
+	}
```

Comment:
indentation is off

Reply:
Compatible with IDE settings and consistent with other CH files. Apparently GO likes it this way